### ISSUE TITLE
iNDS interface design

### ISSUE THREAD

**Issue by WilliamLCobb** [WilliamLCobb/iNDS#119]
As I've said many times, the current overlay design has gotten pretty bulky and changing options requires moving around tons of menus. I would really like to redo the design for the next big release. However, my interface design skills are pretty below average so I'm going to need some help.

We can start with the game overlay. I've created an initial design that looks like this
![image](https://cloud.githubusercontent.com/assets/870904/14972167/3f127970-109f-11e6-935f-c2bd3a30b771.png)

The buttons are pretty ugly but it gets the idea across. I would like to get to any menu or option in just a few touches instead of going back and forth between layers of views like it's set up now. Any simple mockups and ideas would be greatly appreciated, and anyone that contributes a fair amount will be mentioned in the credits of iNDS

**Comment by Pepperoony** [comment]
Why not ask Riley Testhut? Or simply try to mimic the UI from GBA4iOS, which I I believe has an excellent UI!


--


Luke Rooney/ Director of Productions
luke.rooney@servitehs.org(mailto:luke_rooney@icloud.com)/(562)-360-4258(tel:(562)-360-4258)


Trinity Corporation
ServiteHS.org
1952 W La Palma Ave, Anaheim(x-apple-data-detectors://5/2)

**Comment by OkazakiTheOtaku** [comment]
Riley has implemented a skin system within GBA4iOS within the game, which (while being a boon) would be hard to implement.

**Comment by DarknesGaming** [comment]
Hmmmm...That seems like a good idea.
I have an idea! Why don't you add some layouts involving different shaped buttons? We should call InfiniDev for help.

**Comment by pmp174** [comment]
I am going to make some mock ups of the design and post them here

**Comment by pmp174** [comment]
![image](https://cloud.githubusercontent.com/assets/15956420/15001319/8d67646c-1164-11e6-9820-1c259177c88c.png)
![image](https://cloud.githubusercontent.com/assets/15956420/15001321/8db52dc8-1164-11e6-9b3e-894f586aa1cf.png)
![image](https://cloud.githubusercontent.com/assets/15956420/15001320/8db4ded6-1164-11e6-803d-3a3afeecd544.png)
![image](https://cloud.githubusercontent.com/assets/15956420/15001322/8db5d534-1164-11e6-9490-0dd77715a2cc.png)

**Comment by pmp174** [comment]
These are mockups I made on a photo processing app. They are not final and they are rough copies. This is how I believe the design should pan out. I swipe in one direction shows the settings which should all be incorporated under a hamburger menu. The rom interface needs to be made richer. By using a bigger version of the cover art it looks better. This is all rough design and can be improved upon.

**Comment by DarknesGaming** [comment]
This looks good. Can we change the color of the interface from red though? The app itself looks aggressive with the color red.

**Comment by pmp174** [comment]
Yeah I mean this is just a mock up. We can also have the app change color based on the cover art of the last played game or make a setting to change the color.

**Comment by IcYGodZ** [comment]
Can there be an option to actually show the nintendo DS system when playing a game? I mean like how GBA4IOS have a game boy advance system image on portrait mode

**Comment by WilliamLCobb** [comment]
I like that design. I'll have to figure out where to get box art from though. I want to add quick saving though, like ppsspp where you have 5 quick save slots but can also save with a name if you need it longterm

**Comment by pmp174** [comment]
I'm not sure but I believe this is the source provenance gets its box art from https://github.com/OpenVGDB/OpenVGDB

**Comment by pmp174** [comment]
OpenVGDB is a database maintained by the OpenEmu Team and automatically downloads available information and artwork

**Comment by pmp174** [comment]
Any updates on this?

**Comment by DarknesGaming** [comment]
I just finished building iNDS 2.0.0. The initial design has problems with the buttons: No matter what button I press, it closes the ROM. I tried to go into the cheats menu but it closed the ROM instead.
I'll just have to fix it myself.

**Comment by WilliamLCobb** [comment]
I'm going to redo the 2.0 branch. Don't worry about updating the changes or working on the buttons

**Comment by DarknesGaming** [comment]
Oh. Does this mean I created the 2.0.0 branch on my fork for nothing?
I got the initial redesign on there, and I also upgraded the TinyXML files. Now there's only 2 files, and they're both actually TinyXML2. I just renamed them to avoid any problems.
Had to change TiXMLDocument and TiXMLElement to tinyxml2::XMLDocument and tinyxml2::XMLElement

**Comment by DarknesGaming** [comment]
I just sent a pull request. The 2.0.0 branch I made in my fork is based off the master branch. Isn't it just cool?

**Comment by DarknesGaming** [comment]
Since I remade the 2.0.0 branch on my fork, contains some upgrades and has all the bug fixes and commits from the master branch, why don't you just add that to your project? I did do some work on it, and I'm really bored.

**Comment by Anuskuss** [comment]
I hope I can get some spare time too since I would like to create a mockup as well.

**Comment by noah978** [comment]
![indsfinalmenu](https://cloud.githubusercontent.com/assets/26331126/23825165/367a6696-064a-11e7-8c6a-38d06ee82b97.png)

I was thinking along the lines of this design.

**Comment by noah978** [comment]
![indsfinaloverlay](https://cloud.githubusercontent.com/assets/26331126/23825374/15792a96-064e-11e7-8e91-7337352b7aea.png)

I think the game overlay doesn't need nearly as many buttons, and should continue to keep the majority within the pop-up menu. simply adding the save state access during gameplay should suffice. Let me know what you think. Thanks for all the hard work!

**Comment by EternalFlameIV** [comment]
I have a jailed iOS 10.3.2 if anyone needs a testing device.
